Deployment/Jenkins/master/parent-instance.groovy

```diff
@@ -0,0 +1,303 @@
+                      },
+                      {
+                        "ParameterKey": "AdminPubkeyURL",
+                        "ParameterValue": "${env.AdminPubkeyURL}"
+                      },
+                      {
+                        "ParameterKey": "BackupFolder",
+                        "ParameterValue": "${env.BackupFolder}"
+                      },
+                      {
+                        "ParameterKey": "BucketTemplate",
+                        "ParameterValue": "${env.BucketTemplate}"
+                      },
+                      {
+                        "ParameterKey": "Ec2Template",
+                        "ParameterValue": "${env.Ec2Template}"
+                      },
+                      {
+                        "ParameterKey": "ElbTemplate",
+                        "ParameterValue": "${env.ElbTemplate}"
+                      },
+                      {
+                        "ParameterKey": "IamRoleTemplate",
+                        "ParameterValue": "${env.IamRoleTemplate}"
+                      },
+                      {
+                        "ParameterKey": "SecurityGroupTemplate",
+                        "ParameterValue": "${env.SecurityGroupTemplate}"
+                      },
+                      {
+                        "ParameterKey": "TargetVPC",
+                        "ParameterValue": "${env.TargetVPC}"
+                      },
+                      {
+                      "ParameterKey": "BackupCronURL",
+                      "ParameterValue": "${env.BackupCronURL}"
+                      },
+                      {
+                      "ParameterKey": "SubnetId",
+                      "ParameterValue": "${env.SubnetId}"
+                      }
+                    ]
+                   /
+                }
+            }
+        stage ('Prepare AWS Environment') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to delete any active ${CfnStackRoot} stacks... "
+                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
+
+                        aws cloudformation wait stack-delete-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
+                    '''
+                }
+            }
+        }
+        stage ('Launch Jenkins Master Stack') {
```

Comment:
Suggestion-only: Might be worth indicating in the stage's name that this is the master-node plus all dependent services? Maybe `Launch AWS-infra plus Jenkins master`?

Reply:
"Launch Jenkins Master Parent Instance Stack"?